airbyte-integrations/connectors/destination-starburst-galaxy/src/main/java/io/airbyte/integrations/destination/starburst_galaxy/StarburstGalaxyStreamCopier.java

```diff
@@ -0,0 +1,178 @@
+    LOGGER.info("[Stream {}] Creating database schema if it does not exist: {}", streamName, schemaName);
+    sqlOperations.createSchemaIfNotExists(database, schemaName);
+  }
+
+  @Override
+  public void createTemporaryTable() throws Exception {
+    String registerTable = format("""
+                                  CALL system.register_table(schema_name => '%s', table_name => '%s',
+                                  table_location => '%s',
+                                  metadata_file_name => '%s')
+                                  """, schemaName, tmpTableName, getTmpTableLocation(), getTmpTableMetadataFileName());
+    LOGGER.info("[Stream {}] Register table: {}", streamName, registerTable);
+    database.execute(registerTable);
+    LOGGER.info("[Stream {}] Table {} is registered", streamName, tmpTableName);
+  }
+
+  protected abstract String getTmpTableMetadataFileName()
+      throws IOException, InterruptedException;
+
+  @Override
+  public void copyStagingFileToTemporaryTable() {
+    // The tmp table is created directly based on the staging file. So no separate copying step is
+    // needed.
+  }
+
+  /**
+   * Adds newly created source columns to target
+   */
+  private void promoteSourceSchemaChangesToDestination()
+      throws SQLException {
+    List<JsonNode> describeTable = database.queryJsons(format("DESCRIBE %s.%s", schemaName, destTableName));
+    LOGGER.info("[Stream {}] Existing table structure for {}.{} table is {}", streamName, schemaName, destTableName, describeTable);
+    Map<String, String> existingColumns = describeTable.stream().collect(
+        Collectors.toMap(column -> column.get("Column").asText().toLowerCase(ENGLISH), column -> column.get("Type").asText().toLowerCase(ENGLISH)));
+    galaxySchema.columns().forEach(columnMetadata -> {
+      String columnName = columnMetadata.name().toLowerCase(ENGLISH);
+      if (!existingColumns.containsKey(columnName)) {
+        try {
+          String alterTable =
+              format(
+                  "ALTER TABLE %s.%s ADD COLUMN IF NOT EXISTS %s %s",
+                  schemaName,
+                  destTableName,
+                  columnName,
+                  columnMetadata.galaxyIcebergType().getDisplayName());
+          LOGGER.info("[Stream {}] Add column {} : {}", streamName, columnName, alterTable);
+          database.execute(alterTable);
+        } catch (SQLException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+  }
+
+  @Override
+  public String createDestinationTable() throws Exception {
+    if (destinationSyncMode == OVERWRITE) {
+      // Drop existing table to propagate source schema changes
+      String dropTable = format("DROP TABLE IF EXISTS %s.%s", schemaName, destTableName);
+      LOGGER.info("[Stream {}] Dropping destination table: {}", streamName, dropTable);
+      database.execute(dropTable);
+    }
+
+    String fields = galaxySchema.columns().stream()
```

Comment:
Talked in person, to support reserved words can double quote identifiers here.

Reply:
Added support for reserved keywords.